modin/experimental/core/execution/native/implementations/omnisci_on_native/dataframe/dataframe.py

```diff
@@ -1737,11 +1720,18 @@ def _arrow_select(self, exprs):
             The resulting table.
         """
         table = self._execute_arrow()
-        schema = table.schema
 
         new_fields = []
         new_columns = []
 
+        if "__index__" in exprs:
```

Comment:
This is an incorrect way to find the required columns. It doesn't have to be projected into `__index__` column, it might have any name (index can be renamed or moved to a data column). Also, there might be multiple columns referencing rowid. You should check referenced column name with the special name `__rowid__` to find required references.